tests/Monolog/Handler/ProcessHandlerTest.php

```diff
@@ -0,0 +1,206 @@
+{
+	/**
+	 * Dummy command to be used by tests that should not fail due to the command.
+	 *
+	 * @var string
+	 */
+	const DUMMY_COMMAND = 'echo';
+
+	/**
+	 * @covers Monolog\Handler\ProcessHandler::__construct
+	 * @covers Monolog\Handler\ProcessHandler::guardAgainstInvalidCommand
+	 * @covers Monolog\Handler\ProcessHandler::guardAgainstInvalidCwd
+	 * @covers Monolog\Handler\ProcessHandler::write
+	 * @covers Monolog\Handler\ProcessHandler::ensureProcessIsStarted
+	 * @covers Monolog\Handler\ProcessHandler::startProcess
+	 * @covers Monolog\Handler\ProcessHandler::handleStartupErrors
+	 */
+	public function testWriteOpensProcessAndWritesToStdInOfProcess()
+	{
+		$fixtures = array(
+			'chuck norris',
+			'foobar1337'
+		);
+
+		$mockBuilder = $this->getMockBuilder('Monolog\Handler\ProcessHandler');
+		$mockBuilder->setMethods(array('writeProcessInput'));
+		// using echo as command, as it is most probably available
+		$mockBuilder->setConstructorArgs(array(self::DUMMY_COMMAND));
+
+		$handler = $mockBuilder->getMock();
+
+		$handler->expects($this->exactly(2))
+			->method('writeProcessInput')
+			->withConsecutive($this->stringContains($fixtures[0]), $this->stringContains($fixtures[1]));
+
+		/** @var ProcessHandler $handler */
+		$handler->handle($this->getRecord(Logger::WARNING, $fixtures[0]));
+		$handler->handle($this->getRecord(Logger::ERROR, $fixtures[1]));
+	}
+
+	/**
+	 * Data provider for invalid commands.
+	 *
+	 * @return array
+	 */
+	public function invalidCommandProvider()
+	{
+		return [
+			[1337],
+			[''],
+			[null],
+			[fopen('php://input', 'r')],
+		];
+	}
+
+	/**
+	 *
+	 * @dataProvider invalidCommandProvider
+	 * @param mixed $invalidCommand
+	 * @covers Monolog\Handler\ProcessHandler::guardAgainstInvalidCommand
+	 */
+	public function testConstructWithInvalidCommandThrowsInvalidArgumentException($invalidCommand)
+	{
+		$mockBuilder = $this->getMockBuilder('Monolog\Handler\ProcessHandler');
+		$mockBuilder->disableOriginalConstructor();
+		$handler = $mockBuilder->getMock();
+
+		$this->setExpectedException('\InvalidArgumentException');
+		/** @var ProcessHandler $handler */
+		$handler->__construct($invalidCommand);
+	}
+
+	/**
+	 * Data provider for invalid CWDs.
+	 *
+	 * @return array
+	 */
+	public function invalidCwdProvider()
+	{
+		return [
+			[1337],
+			[''],
+			[fopen('php://input', 'r')],
+		];
+	}
+
+	/**
+	 * @dataProvider invalidCwdProvider
+	 * @param mixed $invalidCwd
+	 * @covers Monolog\Handler\ProcessHandler::guardAgainstInvalidCwd
+	 */
+	public function testConstructWithInvalidCwdThrowsInvalidArgumentException($invalidCwd)
+	{
+		$mockBuilder = $this->getMockBuilder('Monolog\Handler\ProcessHandler');
+		$mockBuilder->disableOriginalConstructor();
+		$handler = $mockBuilder->getMock();
+
+		$this->setExpectedException('\InvalidArgumentException');
+		/** @var ProcessHandler $handler */
+		$handler->__construct('fake command', Logger::DEBUG, true, $invalidCwd);
```

Comment:
don't use a partial mock to test the exception. Just perform a normal instantiation. It will be much more readable, and more reliable (as it will test the real thing it want to test)